openedx/core/djangoapps/content/learning_sequences/views.py

```diff
@@ -108,14 +112,20 @@ def _sequence_repr(self, sequence, sequence_schedule, accessible_sequences):
                     'due': sequence_schedule.due,
                 }
 
-            return {
+            sequence_representation = {
                 "id": str(sequence.usage_key),
                 "title": sequence.title,
                 "accessible": sequence.usage_key in accessible_sequences,
                 "inaccessible_after_due": sequence.inaccessible_after_due,
                 **schedule_item_dict,
             }
 
+            # Only include this data if special exams are on
+            if settings.FEATURES.get('ENABLE_SPECIAL_EXAMS', False):
```

Comment:
@pcockwell This doesn't seem to respect the course specific `Enable Timed Exams` setting:
![image](https://user-images.githubusercontent.com/27893385/95467758-353c9e00-0954-11eb-9c75-cae00a96335e.png)

Even after removing a special exam and switching the flag off, I still see this in the course outline API:
```

{
    "entrance_exam_id": null,
    "course_key": "course-v1:OpenCraft+TNL-7264+2020",
    "outline": {
        "sequences": {
            "block-v1:OpenCraft+TNL-7264+2020+type@sequential+block@cf06602ad6b0478b80a2491fc438760f": {
                "accessible": true,
                "effective_start": "2020-10-08T00:00:00Z",
                "inaccessible_after_due": false,
                "id": "block-v1:OpenCraft+TNL-7264+2020+type@sequential+block@cf06602ad6b0478b80a2491fc438760f",
                "start": null,
                "exam": {
                    "short_description": "Timed Exam",
                    "in_completed_state": false,
                    "suggested_icon": "fa-clock-o"
                },
                "title": "Subsection",
                "due": null
            }
        },
        "sections": [
            {
                "start": null,
                "title": "Section",
                "effective_start": "2020-10-08T00:00:00Z",
                "id": "block-v1:OpenCraft+TNL-7264+2020+type@chapter+block@f6de9baadc2444fda2f792d38b67d508",
                "sequence_ids": [
                    "block-v1:OpenCraft+TNL-7264+2020+type@sequential+block@cf06602ad6b0478b80a2491fc438760f"
                ]
            }
        ]
    },
    "at_time": "2020-10-08T13:48:33.507356Z",
    "username": "verified",
    "course_start": "2020-10-08T00:00:00Z",
    "days_early_for_beta": null,
    "published_version": "5f7f1715b07a7bb9a37c4322",
    "self_paced": false,
    "user_id": 7,
    "course_end": null,
    "title": "Content Gating Test",
    "published_at": "2020-10-08T13:41:41.286000Z"
}
```

I don't think this is the desired behavior though, can you fix this?

Reply:
@giovannicimolin

This does appear to be undesired behavior. Though, my mandate for this task was to convert the `[MilestonesAndSpecialExamsTransformer](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_api/blocks/transformers/milestones.py)` into three separate outline processors, and the [current implementation](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/course_api/blocks/transformers/milestones.py#L75-L78) does not compare against or check the course specific `Enabled Timed Exams` flag.

@ormsbee @symbolist Can you comment on this behavior? I am happy to fix this if this result is undesired.

Reply:
@pcockwell: Please leave it in to match MilestonesAndSpecialExamsTransformer's behavior with a comment indicating that.